Tools/clinic/clinic.py

```diff
@@ -828,6 +828,31 @@ class CLanguage(Language):
             #define {methoddef_name}
         #endif /* !defined({methoddef_name}) */
     """)
+    DEPRECATED_POSITIONAL_PROTOTYPE: Final[str] = r"""
+        #if PY_MAJOR_VERSION == {major} && \
+            PY_MINOR_VERSION == {minor} && \
+            PY_RELEASE_LEVEL == PY_RELEASE_LEVEL_ALPHA
+        #  ifdef _MSC_VER
+        #    pragma message ("{cpp_warning}")
+        #  else
+        #    warning "{cpp_warning}"
+        #  endif
+        #elif PY_MAJOR_VERSION > {major} || \
+             (PY_MAJOR_VERSION == {major} && PY_MINOR_VERSION > {minor}) || \
+             (PY_MAJOR_VERSION == {major} && \
+              PY_MINOR_VERSION == {minor} && \
+              PY_RELEASE_LEVEL == PY_RELEASE_LEVEL_GAMMA)
```

Comment:
Would not be simpler to use PY_VERSION_HEX?
```suggestion
        #elif PY_VERSION_HEX >= 0x{major:02x}{minor:02x}00C0
```
The first `#if` condition can be simpler if you change the order of `#if`s:
```
        #if PY_VERSION_HEX >= 0x{major:02x}{minor:02x}00C0
        // error
        #elif PY_VERSION_HEX >= 0x{major:02x}{minor:02x}00A0
        // warning
```

Reply:
Good point! Thanks.